cmd/server/pactasrv/authz.go

```diff
@@ -79,15 +79,27 @@ func notFoundOrUnauthorized[T ~string](actorInfo actorInfo, action pacta.AuditLo
 }
 
 func (s *Server) auditLogIfAuthorizedOrFail(ctx context.Context, status *authzStatus) error {
+	zapFields := []zap.Field{
```

Comment:
You should make this a function, [like here](https://github.com/RMI/opgee-api/blob/d342c0c21c0dbf4c62025a4ee71a74094f4f37f0/cmd/apiserver/server/opgee.go#L252-L257). The reason is that Go has this very subtle footgun with [`append`](https://pkg.go.dev/builtin#append) where it may or may not modify the slice passed in.

In this case, it doesn't matter because every time you append you also immediately return, but still, some linters would call this out.

Reply:
Oh really smart.